plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -72,15 +87,66 @@ rules :: Recorder (WithPriority Log) -> PluginId -> Rules ()
 rules recorder plId = do
   define (cmapWithPrio LogShake recorder) $
     \GetStanDiagnostics file -> do
-      config <- getPluginConfigAction plId
-      if pluginEnabledConfig plcDiagnosticsOn config then do
+      plugConfig <- getPluginConfigAction plId
+      if pluginEnabledConfig plcDiagnosticsOn plugConfig then do
           maybeHie <- getHieFile file
           case maybeHie of
             Nothing -> return ([], Nothing)
             Just hie -> do
-              let enabledInspections = HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)]
-              -- This should use Cabal config for extensions and Stan config for inspection preferences is the future
-              let analysis = runAnalysis Map.empty enabledInspections [] [hie]
+              let currentHSfromHIEAbs = hie_hs_file hie
+              currentHSfromHIERel <- liftIO $ makeRelativeToCurrentDirectory currentHSfromHIEAbs
+              -- This codes follows what 'runStan' does, from the module 'Stan'
+
+              -- There aren't any cli args. isLoud=False=Silent output
+              let isLoud = False -- Should this be enabled when debugging? Enables default stan cli output
+              let stanArgsConfigFile = Nothing -- There aren't any cli args
+
+              EnvVars{envVarsUseDefaultConfigFile} <- liftIO getEnvVars
+              logWith recorder Debug (LogDebug $
+                "envVarsUseDefaultConfigFile: " <> (T.pack $ show envVarsUseDefaultConfigFile))
+
+              let defConfTrial = envVarsUseDefaultConfigFile -- There aren't any cli args: <> stanArgsUseDefaultConfigFile
+              let useDefConfig = maybe True snd (trialToMaybe defConfTrial)
+
+              tomlConfig <- liftIO $ getTomlConfig isLoud useDefConfig stanArgsConfigFile
+              let configTrial = finaliseConfig $ defaultConfig <> tomlConfig -- There aren't any cli args: <> stanArgsConfig
+              logWith recorder Debug (LogDebug $ "Final stan config result\n" <> ( T.pack $ show configTrial))
+
+              (cabalExtensionsMap, checksMap, confIgnored) <- case configTrial of
+                   FiascoL es -> do
+                      logWith recorder Warning (LogWarnConf es)
+                      pure (Map.empty,
+                            HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)],
+                            [])
+                   ResultL warnings stanConfig -> do
+                      -- I'm not sure this is the best way to obtain the .cabal
```

Comment:
Again, ideally stan would expose a function for working out everything it needs given a current directory. We've increasingly found that it's risky to deviate from what a tool does when run at the CLI. For example, by doing this maybe we do a better job of finding the cabal file than the stan code would, so the user will see different results in HLS and the stan CLI. Better to get stan to give us a function that just sets everything up given a working directory, and then we can just call that.

Reply:
Maybe we could define the function here that we'd like stan to provide, that gives them something to look at!